pkg/apitype/core.go

```diff
@@ -86,6 +86,18 @@ type DeploymentV2 struct {
 	Manifest ManifestV1 `json:"manifest" yaml:"manifest"`
 	// Resources contains all resources that are currently part of this stack after this deployment has finished.
 	Resources []ResourceV2 `json:"resources,omitempty" yaml:"resources,omitempty"`
+	// InFlightOperations are all operations that were known by the engine to be currently executing.
+	InFlightOperations []OperationV1 `json:"in_flight_operations,omitempty" yaml:"in_flight_operations,omitempty"`
+}
+
+// OperationV1 represents an operation that the engine is performing. It consists of a Resource, which is the state
+// that the engine used to initiate the operation, and a Status, which is a string representation of the operation
+// that the engine initiated.
+type OperationV1 struct {
+	// Resource is the state that the engine used to initiate this operation.
+	Resource ResourceV2 `json:"resource" yaml:"resource"`
+	// Status is a string representation of the operation that the engine is performing.
+	Status string `json:"status" yaml:"status"`
```

Comment:
Is it possible for this to be an enum type, e.g. `ResourceStatusDelete`, `ResourceStatusReplace`? It is super-lame that we need to duplicate enums found in the engine layer in `apitypes`, but it's the only way that the service can assert the meaning of various `Status` values.

Reply:
Yep, I can do that. I would kind of like to keep them as strings so that their meanings are obvious when viewed in the context of an exported deployment, but I can add the apitype definitions here.